File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/c++/m-doce.cpp
#include <iostream>

int main()
{
    //Operadores aritméticos
    printf("%d \n", 5 + 7); //Suma
    printf("%d \n", 15 - 8); //Resta
    printf("%d \n", 3 * 9); //Multiplicación
    printf("%.3f \n", 99 / 7.0); //División
    printf("%d \n", 18 % 5); //Módulo - Devuelve el resto de la división (en este caso, 3)

    //Operadores de asignación y comparación
    int num = 100; //El '=' se usa para asignar valores
    num == 50? printf("Yes\n") : printf("No\n"); //El '==' se usa para comparar la igualdad entre valores
    num != 50? printf("Yes\n") : printf("No\n"); //El '!=' se usa para comparar la desigualdad entre valores
    num < 50? printf("Yes\n") : printf("No\n"); //El '<' devuelve true cuando el valor a su izq es menor
    num > 50? printf("Yes\n") : printf("No\n"); //El '>' devuelve true cuando el valor a su izq es mayor

    //Operadores lógicos
    ((10 % 2) == 0) and ((20 % 3) == 0)? printf("Yes\n") : printf("No\n"); //El 'and' o '&&' se usa para evaluar que dos o más condiciones se cumplan 
    ((10 % 3) == 0) or ((20 % 3) == 0)? printf("Yes\n") : printf("No\n"); //El 'or' o '||' se usa para evaluar que al menos una condición se cumpla
    not ((10 % 3) == 0) and not ((20 % 3) == 0)? printf("Yes\n") : printf("No\n"); //El 'not' o '!' se usa para invertir la evaluación de una condición

    //Estructuras de control

    //If-Else, se usa para bifurcar el código en base a una condición
    if(4 > 5)
    printf("Se cumple la condicion\n");
    else
    printf("No se cumple la condicion\n");

    //Switch, se usa para ejecutar una acción en base al valor de una variable (útil para los menúes)
    int diaDeLaSemana = 5;
    switch(diaDeLaSemana)
    {
        case 1:
        printf("Lunes\n");
        break;

        case 2:
        printf("Martes\n");
        break;

        case 3:
        printf("Miércoles\n");
        break;

        case 4:
        printf("Jueves\n");
        break;

        case 5:
        printf("Viernes\n");
        break;

        case 6:
        printf("Sábado\n");
        break;

        case 7:
        printf("Domingo\n");
        break;
    }

    //Estructuras de repetición

    //For, sirve para ejecutar instrucciones en bucle una cantidad de veces determinada
    for(int i=0; i<10; i++)
    {
        printf("%d\n", i);
    }

    //While, sirve para ejecutar instrucciones en bucle una cantidad de veces indeterminada (por lo general hasta cumplir cierta condición)
    int userInput = 0;
    printf("Ingrese un numero para ver su doble, o '0' para finalizar: ");
    scanf("%d", &userInput);
    while(userInput != 0)
    {
        printf("%d x 2 = %d\n", userInput, (userInput*2));
        printf("Ingrese un numero para ver su doble, o '0' para finalizar: ");
        scanf("%d", &userInput);
    }

    // - - - - - SEGUNDO EJERCICIO - - - - -
    printf("\n\n [*] Ejercicio extra [*]\n\n");

    for(int i=10; i<56; i++)
    {
        if(((i % 2) == 0) and ((i % 3) != 0) and (i != 16))
        {
            printf("[+] %d\n", i);
        }
    }

    return 0;
}

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/m-doce.py
#Operadores aritméticos: +, -, *, /, %
suma = (7 + 5)
resta = (10 - 8)
multiplicacion = (3 * 9)
division = (21 / 2)
divisionEntera = (21 // 2) #Devuelve solo la parte entera de la división
modulo = (21 % 2) #Devuelve el resto de la división (en este caso, 1)
potencia = (2 ** 5) #Imprime el numero de la izquierda elevado al de la derecha

#Operadores de asignación y comparación: =, !=, <, >, ==
asignacion = 1 #El simbolo '=' se usa para asignar valores a las variables
print(9==3) #El '==' se utiliza para comparar la igualdad entre valores (devolviendo verdadero o falso)
print(9!=3) #El '!=' se utiliza para comparar si existe diferencia entre dos valores (devolviendo verdadero o falso)
print(9<3) #El '<' devuelve true cuando el valor a su izq es menor
print(9>3) #El '>' devuelve true cuando el valor a su izq es mayor

#Operadores lógicos: and, or, not
print(True and False) #Con 'and' evaluamos si se cumplen dos o más condiciones
print(True or False) #Con 'or' evaluamos si se cumple al menos una condición
print(not True) #Con 'not' negamos/revertimos una condición

#Operadores de identidad: is, is not
#Estos operadores se utilizan para saber si dos variables apuntan o no al mismo objeto en memoria
first = 10
second = 10
third = first
print(first is second)
print(first is not second)
print(first is third)
print(first is not third)

#Operadores de pertenencia: in, not in
#Estos operadores se utilizan para saber si un valor esta presente en un conjunto de datos (string, arrays, etc)
palabra = "Operadores"
print("u" in palabra)
print(not "u" in palabra)

#Estructuras de control
#If-Else, se utiliza para seguir por un camino u otro en base a una condición
number = 10
if number > 10:
    print("El numero es mayor que 10")
elif number == 10:
    print("El numero es 10")
else:
    print("El numero es menor que 10")

#Estructuras de repetición
#For, se utiliza para ejecutar instrucciones en bucle una cantidad de veces determinada

for i in range(10):
    print(i)

#While, se utiliza para ejecutar instrucciones en bucle una cantidad de veces indeterminada (por lo general hasta cumplir cierta condición)
userInput = int(input("Ingrese un numero para ver su doble, o '0' para finalizar: "))
while userInput != 0:
    print(userInput*2)
    userInput = int(input("Ingrese un numero para ver su doble, o '0' para finalizar: "))

#Estructuras de excepciones
try:
    print(19 / 0)
except:
    print("Se ha producido un error")
finally:
    print("Fin del manejo de excepciones")

#EJERCICIO EXTRA

for value in range(10, 56):
    if (value%2 == 0) and (value%3 != 0) and (value != 16):
        print(value)

print("Fin del programa")